docs/herbarium/working-with-herbarium.md

```diff
@@ -0,0 +1,139 @@
+- `initialPath`: (optional) default `process.cwd()`
+- `avoidFiles`: (optional) default `(fileName) => (fileName.includes('.spec.js') || fileName.includes('.test.js') ? false : fileName)`
+- `onlySpecs`: (optional) default `(fileName) => (fileName.includes('.spec.js') ? fileName : false)`
+- `entitiesPath`: (optional) default `/src/domain/entities`
+- `usecasesPath`: (optional) default `/src/domain/usecases`
+- `specPath`: (optional) default `/src/domain/`
+- `repositoriesPath`: (optional) default `/src/infra/repositories`
+
+### Adding Objects and Metadata
+
+#### Entities
+
+```javascript
+// src/domain/entities/item.js
+const { entity, field } = require('@herbsjs/herbs')
+const { herbarium } = require('@herbsjs/herbarium')
+
+const Item =
+    entity('Item', {
+        ...
+    })
+
+module.exports =
+    herbarium.entities
+        .add(Item, 'Item')
+        .entity
+```
+
+The second parameter of the `herbarium.entities.add` function is the entity id for herbarium. It is optional and if none is provided, it uses the entity name (`Item`).
+
+#### Use Cases
+
+```javascript
+// src/domain/usecases/item/createItem.js
+const { usecase } = require('@herbsjs/herbs')
+const { herbarium } = require('@herbsjs/herbarium')
+const { Item } = require('../entities/item')
+
+const createItem = (injection) =>
+    usecase('Create Item', {
+        ...
+    })
+
+module.exports =
+    herbarium.usecases
+        .add(createItem, 'CreateItem')
+        .metadata({ group: 'Items', operation: herbarium.crud.create, entity: Item })
+        .usecase
+```
+
+The second parameter of the `herbarium.usecases.add` function is the usecase id for herbarium. It is optional and if none is provided, it uses the usecase description (`Create Item`).
+
+```javascript
+// src/domain/usecases/createItem.spec.js
+const { spec, scenario, given, check } = require('@herbsjs/aloe')
+
+const createItemSpec = spec({ 
+    ...
+    })
+
+module.exports =
+    herbarium.specs
+        .add(createItemSpec, 'CreateItemSpec')
+        .spec
+```
+
+The second parameter of the `herbarium.specs.add` function is the entity id for herbarium.
+
+ #### Repository
+
+```javascript
+const { herbarium } = require('@herbsjs/herbarium')
+const { Repository } = require("@herbsjs/herbs2knex")
+const { Item } = require('../../../domain/entities/item')
+
+class ItemRepository extends Repository {
+    constructor(injection) {
+        ...
+    }
+}
+
+module.exports =
+    herbarium.repositories
+        .add(ItemRepository, 'ItemRepository')
+        .metadata({ entity: Item })
+        .repository
+```
+
+The second parameter of the `herbarium.repositories.add` function is the repository id for herbarium. It is optional and if none is provided, it uses the repository class name (`ItemRepository`).
+
+### Consuming Objects
+
+#### All
+
+```javascript
+const entities = herbarium.entities.all
+const usecases = herbarium.usecases.all
+const repositories = herbarium.repositories.all
+```
+
+#### findBy
```

Comment:
Add description. Ex: `Find entities matched by the filter`

Also must describe / doc how the "query language" works.

Reply:
"All operations you can do on any data can be boiled down to Create, Read, Update, and Delete (CRUD). You can create something new, you can read it, update it, and finally delete it if you wish. For that, you can assign the property that best fits your use case and then filter it through the method findBy()"
Is it ok for me to explain it this way?